contrib/IECoreUSD/include/IECoreUSD/SceneAlgo.h

```diff
@@ -0,0 +1,64 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2021, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef IECOREUSD_SCENEALGO_H
+#define IECOREUSD_SCENEALGO_H
+
+#include "IECoreUSD/Export.h"
+
+#include "IECoreScene/SceneInterface.h"
+
+IECORE_PUSH_DEFAULT_VISIBILITY
+#include "pxr/base/tf/token.h"
+IECORE_POP_DEFAULT_VISIBILITY
+
+namespace IECoreUSD
+{
+
+namespace SceneAlgo
+{
+
+IECOREUSD_API pxr::TfToken internalRootNameToken();
+IECOREUSD_API IECore::InternedString internalRootName();
+IECOREUSD_API IECoreScene::SceneInterface::Path fromInternalPath( const IECoreScene::SceneInterface::Path& scenePath );
+IECOREUSD_API IECoreScene::SceneInterface::Path toInternalPath( const IECoreScene::SceneInterface::Path& scenePath );
+IECOREUSD_API std::string fromInternalName( const IECore::InternedString& name );
+IECOREUSD_API std::string toInternalName( const IECore::InternedString& name );
+
+} // namespace SceneAlgo
+
+} // namespace IECoreUSD
```

Comment:
What's the reason for this being public? Am I right in thinking it's only really useful inside the scene cache file format (and the tests?). As a casual reader I would have no real idea what any of these functions are for or how I should use them. I don't expect anyone outside of Image Engine would need the `.scc` loading, and calling it something as generic as `SceneAlgo` doesn't make this limited scope very clear. Ideally I think we'd hide it, but perhaps a decent minimum would be to choose a more descriptive namespace and add a brief explanation of what it is for?